packages/babel-generator/src/generators/base.ts

```diff
@@ -25,21 +32,22 @@ export function BlockStatement(this: Printer, node: t.BlockStatement) {
   this.token("{");
   this.printInnerComments(node);
 
-  const hasDirectives = node.directives?.length;
-
-  if (node.body.length || hasDirectives) {
-    this.newline();
-
-    this.printSequence(node.directives, node, { indent: true });
-    if (hasDirectives) this.newline();
-
-    this.printSequence(node.body, node, { indent: true });
-    this.removeTrailingNewline();
-
-    if (!this.endsWith(charCodes.lineFeed)) this.newline();
+  const newline = node.body.length ? 2 : 1;
+  const len = node.directives?.length || 0;
```

Comment:
Nit: move `newline` inside the `if` statement, and rename `len` to `directivesLen`. I was initially confused to why we would need a newline in an empty block, until I realized that it's because the block contains directives.

Also, `|| 0` is not needed.